module/extension/Director/src/director/challenge_priority.cpp

```diff
@@ -106,6 +107,7 @@ namespace module::extension {
         // This can happen when we are checking tasks with a `Needs` relationship against a task higher in the tree and
         // we already own that task. In that case we say that the challenger wins because we already own it and can
         // replace it.
+        // TLDR parent beats child
```

Comment:
yikes

Reply:
I thought the same thing as I wrote the comment :P